tests/tools.py

```diff
@@ -163,6 +166,24 @@ def emulate_assertIsNotNone(self, x, msg=None):
             msg = msg if msg is not None else '{0} is None'.format(_safe_repr(x))
             self.fail(msg)
 
+    def emulate_raises_regex(self, exception_type, regex, function, *args, **kwargs):
+        if PY_VERSION_MAJOR == 3 and PY_VERSION_MINOR == 6:
```

Comment:
But I thought Python 3.6 and higher already implemented assertRaisesRegex? If that's the case, then emulate_raises_regex will never be called (because it won't be patched into cls), so there's no need for this code at all.

I am assuming that the signature for assertRaisesRegex is unchanged in 3.6 and 3.7, and that emulate_raises_regex implements exactly that signature.

Reply:
Sorry, I mucked with the version numbers while testing and forgot to change it back.  It's 3.6 vs 2.7.  I've removed the code for 3.6 as it's not needed as you said.  That said, the versions still aren't correct and I'm going to fix that though I don't have every version of python to test, so I'm going to go off the documentation.

---
In reply to: [207088817](https://github.com/Azure/WALinuxAgent/pull/1281#discussion_r207088817) [](ancestors = 207088817)

Reply:
Ok, fixed the versions based on what I've read.  assertRaisesRegexp was introduced in 2.7 and was replaced by assertRaisesRegex in 3.2.  I'm not going to guess at python 4, but technically this method would still work.

---
In reply to: [207311754](https://github.com/Azure/WALinuxAgent/pull/1281#discussion_r207311754) [](ancestors = 207311754,207088817)